packages/terra-tag/src/Tag.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Tag.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  TAB: 9,
+};
+
+const tagTheme = {
+  NORMAL: 'normal',
+};
+
+const propTypes = {
+  /**
+   * Sets the href. When set will render the component as an anchor tag.
+   */
+  href: PropTypes.string,
+  /**
+   * An optional icon.
+   */
+  icon: PropTypes.element,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Callback function triggered when tag loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Callback function triggered when tag gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
+   */
+  onKeyUp: PropTypes.func,
+  /**
+   * Sets the tag text.
+   */
+  text: PropTypes.string.isRequired,
+  /**
+   * Sets the tag theme.
+   */
+  theme: PropTypes.string,
+};
+
+const defaultProps = {
+  text: '',
```

Comment:
This probably isn't a huge deal, but it feels a little weird to have a default value for a required feel. What do you think?

Reply:
Yes  I agree , removed it [here](https://github.com/cerner/terra-core/pull/1306/commits/0c5264f62462ae9bc4cbedbae83c5cac42c28580) .